### ISSUE TITLE
Cannot parse clojure.main (ns-qualitified `:keys: syntax)

### ISSUE THREAD

**Issue by pesterhazy** [clojure-lsp/clojure-lsp#135]
```
WARN  clojure-lsp.parser: params nil [{:as triage-data, :clojure.error/keys [phase source path line column symbol class cause spec]}]
WARN  clojure-lsp.parser: Cannot parse jar:file:///Users/user/.m2/repository/org/clojure/clojure/1.10.1/clojure-1.10.1.jar!/clojure/main.clj
 nil
 (defn ex-str
  "Returns a string from exception data, as produced by ex-triage.
  The first line summarizes the exception phase and location.
  The subsequent lines describe the cause."
  {:added "1.10"}
  [{:clojure.error/keys [phase source path line column symbol class cause spec]
    :as triage-data}]
```

(cut the rest)

This fails to parse because of the unusual `{:clojure.error/keys [phase source path line column symbol class cause spec]}` shortcut destructuring syntax.

```
user=> (let [{:myns/keys [foo]} {:myns/foo 123}] foo)
123
```

**Comment by snoe** [comment]
Interesting - I think your example should actually work, the problem seems to be using the `:ns/keys` destructuring while shadowing a clojure builtin (`symbol`,`spec`) in `ex-str`

**Comment by dpsutton** [comment]
I'm unable to trigger a failing case somehow:

```clojure
(let [code "(let [{:clojure.error/keys [phase source path line column symbol class cause spec]} {}])"]
    (parser/find-usages code :clj {}))
```

I also see logs that clojure-lsp can't parse the Datafiable protocol but I was unable to reproduce that either

```
WARN  clojure-lsp.parser: Cannot parse jar:file:///Users/dan/.m2/repository/org/clojure/clojure/1.10.1/clojure-1.10.1.jar!/clojure/core/protocols.clj 
 nil 
 (defprotocol Datafiable
  :extend-via-metadata true

  (datafy [o] "return a representation of o as data (default identity)"))
```

**Comment by ericdallo** [comment]
Tested and it works now, this was fixed after we merged #261 :)
